### DISCUSSION TITLE
Missing link to bare-metal project: CoreAXI4DMAController

### DISCUSSION THREAD

**Discussion by tsibbett** [polarfire-soc/polarfire-soc-documentation#192] [Bare metal embedded software] [answered]
(I emailed a contact in technical support with regards to this issue - putting it here for reference)

In the mpfs-axi4-stream-demo readme, there is a broken link to a bare-metal project CoreAXI4DMAController which will initiate the data stream and subsequently verify the data. Do you have a copy of that project that I can use until the link is fixed?

**Comment by gsriker77** [comment]
Can you please let us know case no / to whom you contacted from Tech support regarding this issue?
So that we can update @same SF portal for Baremetal example. We can't share directly here.

Thanks,

**Reply by tsibbett** [reply]
I spoke with Cameron Wade initially, but he forwarded the question to someone in support who indicated that the link is fixed internally but has not yet been pushed. I was just hoping to get access in the interim until the link is fixed.

**Comment by gsriker77** [comment]
Can it possible to raise new Techsupport case for it? We will share it in case itself.

**Reply by tsibbett** [reply]
Sure - it is case number 01063550. Thanks!

**Comment by gsriker77** [comment, marked as answer]
Shared required information in Tech support case.

**Reply by cole-maxwell** [reply]
Could I please have interim access to the bare metal project as well?

**Reply by hughbreslin** [reply]
Hi Cole, please open a case :)

**Reply by adaneyn** [reply]
Hi Cole did they give you interim access?

**Comment by cole-maxwell** [comment]
The link to the bare metal project for the CoreAXI4DMAController is still missing.
Will this be added?
The documentation for the demo links here: https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/fpga-ip/CoreAXI4DMAController

**Reply by hughbreslin** [reply]
Hi Cole, yes this will be released

**Reply by Shivaraj199** [reply]
Hello @hughbreslin,
I am not able get CoreAXI4DMAController bare metal code, Can you please send me the link so I can access the code 
Thank you

**Comment by gsriker77** [comment]
Sent to tsibbett@a2etechnologies.com. pl confirm.

**Reply by cole-maxwell** [reply]
Hi, could you please send it to my email? It's cole.h.maxwell@gmail.com

**Reply by Shivaraj199** [reply]
Hello the baremetal code link is still broken, when will it be updated?

**Comment by nitindeshpande** [comment]
Hello,
 Apologies for the delay. The project is being updated. It will be available in the repository in next 2 weeks.

**Reply by adaneyn** [reply]
Broken promises. I would like to know what is DMA base address in this project. Since DMA bench marking exists I may be able to reuse that code for the AXI DMA. Please share some line of code on how to read and write to the AXI4 DMA. Thanks

**Comment by bilievski** [comment]
Hi all - seems that the link to CoreAXI4DMAController files is still broken: https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/fpga-ip/CoreAXI4DMAController

Any updates on this?

**Reply by gsriker77** [reply]
We are restructuring github site for polarfire-soc. Soon it will get updated in link.
Meanwhile, can you please raise Microchip Salesforce case to share it

**Reply by adaneyn** [reply]
Well received thank you!

**Reply by gsriker77** [reply]
You can download from below github link :

https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/bbe684720f1eb762d3851ae90592d320b1a03b45/applications/benchmarks/dma_benchmarking/mpfs-dma-benchmarking/src/platform/drivers/fpga_ip/CoreAXI4DMAController
